endpoints/openrtb2/auction.go

```diff
@@ -1448,7 +1449,13 @@ func getStoredRequestId(data []byte) (string, bool, error) {
 		return "", true, errors.New("ext.prebid.storedrequest.id must be a string")
 	}
 
-	return string(value), true, nil
+	// Generates Request ID if flag is set to true -- used to solve issue with amp auctions
```

Comment:
This is supposed to return the ID of the stored request, not the openrtb request's ID. Also by generating a random stored request ID you are virtually guaranteeing that the stored request lookup will fail, and potentially hammering the database with failing lookups.

Reply:
To avoid having to unmarshal/remarshal the stored request json, you probably want to patch the retrieved stored request with `{  "id": "<generated uuid>" }` before applying the patch to the incoming request. Unless we want the spec that the flag should always overwrite any incoming request IDs, in which case we will have to do something else. But I don't see a use case for overwriting the request ID in the incoming request other than the case where a PBS instance is only supporting a client that always sends a non-random ID, and no other clients. But in this case it really should be fixed on the client side rather than hacking PBS to enforce unique IDs in all cases.

Reply:
Just to keep this PR updated, Hans and I talked about these comments and what I need to do to properly address them! A commit will be pushed today or tomorrow with the proper fixes / tests.